geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/FunctionCommandsDUnitTest.java

```diff
@@ -37,649 +25,203 @@
-        for (int i = 0; i < 10; i++) {
-          region.put("key" + (i + 200), "value" + (i + 200));
-        }
+    server2.invoke(() -> {
+      Cache cache = LocatorServerStartupRule.serverStarter.getCache();
+      RegionFactory<Integer, Integer> dataRegionFactory =
+          cache.createRegionFactory(RegionShortcut.PARTITION);
+      Region region = dataRegionFactory.create(REGION_ONE);
+      for (int i = 0; i < 10000; i++) {
+        region.put("key" + (i + 400), "value" + (i + 400));
+      }
+      region = dataRegionFactory.create(REGION_TWO);
+      for (int i = 0; i < 10; i++) {
+        region.put("key" + (i + 200), "value" + (i + 200));
       }
     });
-  }
 
-  SerializableRunnable checkRegionMBeans = new SerializableRunnable() {
-    @Override
-    public void run() {
-      final WaitCriterion waitForManagerMBean = new WaitCriterion() {
-        @Override
-        public boolean done() {
-          final ManagementService service = ManagementService.getManagementService(getCache());
-          final DistributedRegionMXBean bean =
-              service.getDistributedRegionMXBean(Region.SEPARATOR + REGION_ONE);
-          if (bean == null) {
-            return false;
-          } else {
-            getLogWriter()
-                .info("Probing for checkRegionMBeans testExecuteFunctionOnRegionBug51480 finished");
-            return true;
-          }
-        }
-
-        @Override
-        public String description() {
-          return "Probing for testExecuteFunctionOnRegionBug51480";
-        }
-      };
-      waitForCriterion(waitForManagerMBean, 2 * 60 * 1000, 2000, true);
-      DistributedRegionMXBean bean = ManagementService.getManagementService(getCache())
-          .getDistributedRegionMXBean(Region.SEPARATOR + REGION_ONE);
-      assertNotNull(bean);
-    }
-  };
-
-  @Test
-  public void testExecuteFunctionOnRegionBug51480() {
-    setupForBug51480();
+    registerFunction(new TestFunction(true, TestFunction.TEST_FUNCTION1), locator, server1,
+        server2);
+    registerFunction(new TestFunction(true, TestFunction.TEST_FUNCTION_RETURN_ARGS), locator,
+        server1, server2);
 
-    // check if DistributedRegionMXBean is available so that command will not fail
-    final VM manager = Host.getHost(0).getVM(0);
-    manager.invoke(checkRegionMBeans);
+    locator.invoke(() -> {
+      Cache cache = LocatorServerStartupRule.locatorStarter.getLocator().getCache();
+      ManagementService managementService = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsMXBean = managementService.getDistributedSystemMXBean();
 
-    final Function function = new TestFunction(true, TestFunction.TEST_FUNCTION1);
-    Host.getHost(0).getVM(0).invoke(new SerializableRunnable() {
-      public void run() {
-        FunctionService.registerFunction(function);
-      }
+      await().atMost(120, TimeUnit.SECONDS).until(() -> dsMXBean.getMemberCount() == 3);
```

Comment:
do we use the beans in some tests that we need to make sure these beans are ready? I am asking because in most of our test, we just connect directly after we started the locator. Hasn't needed to implement this wait yet.

Reply:
We don't directly use the beans, but sometimes the function execution can return `no members` so this is just ensuring that all members are actually ready.